widgets/htdocs/components/95_AFDB.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+ * Copyright [2016-2017] EMBL-European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+Ensembl.Panel.AFDB = Ensembl.Panel.Content.extend({
+
+  init: function() {
+    this.base.apply(this, arguments);
+
+    var ensemblAlphafoldElement = document.querySelector('ensembl-alphafold-viewer'); // <-- custom element will be included in server response
+
+    if (!ensemblAlphafoldElement) {
+      this.onMissingData();
```

Comment:
The check in the `if` statement is about the presence of the element but the function `this.onMissingData()` is about "No data available". is that the right message to put here  if the widget doesnt show?

Reply:
You will see [here](https://github.com/Ensembl/public-plugins/blob/1c9cff5342ba6dfa018dab2dd294c01f3e01aaef/widgets/modules/EnsEMBL/Web/Component/Transcript/AFDB.pm#L40) that the alphafold element will not be present on the page if there is no translation_object associated with this feature. This is a case of missing data.

Reply:
i dont think the code will reach here if the data isn't present from the perl side. If you look at the function (the one you linked to)  it return if it doesnt find any data which means the panel below wont even be loaded which contains the js file for this. I dont think you need that check here (jsut do this.loadScript). if you want to put the check it will have to be on the perl side.

Reply:
@azangru not sure if you seen my comment above.

Reply:
Ok, I checked what will happen if somebody tries to load alphafold view for a non-coding transcript (which wouldn't make sense from the data perspective). Turns out, perl code can already handle this just fine. Example: http://ves-hx2-75.ebi.ac.uk:8450/Arabidopsis_thaliana/Transcript/AFDB?db=core;g=AT1G04017;r=1:89235-108903;t=AT1G04017.1

I'm removing this check from the javascript side and assuming that if 95_AFDB.js is executed, it means that the `ensembl-alphafold-viewer` component is always present.

Reply:
The availability call fails too (ie the LH menu is greyed out)